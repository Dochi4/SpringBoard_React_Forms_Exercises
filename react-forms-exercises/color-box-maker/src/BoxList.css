.container {
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(200px, 1fr));
  gap: 16px;
  background-color: rgb(129, 133, 136);
  padding: 20px;
  box-sizing: border-box;
}

.container > div {
  border: 1px solid black;
  padding: 10px;
  border-radius: 8px;
  box-shadow: 2px 2px 6px rgba(0, 0, 0, 0.2);
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-between;
}
