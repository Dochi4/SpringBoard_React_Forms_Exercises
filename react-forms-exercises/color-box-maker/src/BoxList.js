import React, { useState } from "react";
import { v4 as uuid } from "uuid";
import NewBoxForm from "./NewBoxForm";
import Box from "./Box";
import "./BoxList.css";

const BoxList = () => {
  const initialState = [
    { id: uuid(), width: "200px", height: "100px", background: "green" },
    { id: uuid(), width: "100px", height: "200px", background: "blue" },
  ];

  const [boxes, setBoxes] = useState(initialState);
  const addBox = (newBoxData) => {
    setBoxes((boxes) => [...boxes, { ...newBoxData, id: uuid() }]);
  };
  const handleRemoveItem = (idToRemove) => {
    setBoxes((boxes) => boxes.filter((box) => box.id !== idToRemove));
  };

  return (
    <div>
      <NewBoxForm addBox={addBox} />
      <div className="container">
        {boxes.map(({ id, width, height, background }) => (
          <Box
            id={id}
            key={id}
            width={width}
            height={height}
            background={background}
            removeBox={handleRemoveItem}
          />
        ))}
      </div>
    </div>
  );
};

export default BoxList;
