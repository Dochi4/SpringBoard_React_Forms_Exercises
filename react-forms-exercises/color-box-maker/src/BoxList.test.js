import React from "react";
import { render, fireEvent } from "@testing-library/react";
import BoxList from "./BoxList";
import TEST_BOX from "./_testCommon";

it("should render without crashing", function () {
  render(<BoxList />);
});

it("matches snapshot", function () {
  const { asFragment } = render(<BoxList />);
  expect(asFragment()).toMatchSnapshot();
});

it("should change the width input", function () {
  const { getByLabelText, getByText, getAllByText } = render(<BoxList />);

  fireEvent.change(getByLabelText("Width (px)"), {
    target: { value: parseInt(TEST_BOX.width) },
  });
  fireEvent.change(getByLabelText("Height (px)"), {
    target: { value: parseInt(TEST_BOX.height) },
  });
  fireEvent.change(getByLabelText("Background Color"), {
    target: { value: TEST_BOX.background },
  });

  fireEvent.click(getByText("Add Box"));

  expect(getAllByText("Remove").length).toBe(3);
});

it("should test Remove", function () {
  const { getAllByText } = render(<BoxList />);

  expect(getAllByText("Remove").length).toBe(2);

  fireEvent.click(getAllByText("Remove")[0]);

  expect(getAllByText("Remove").length).toBe(1);
});
