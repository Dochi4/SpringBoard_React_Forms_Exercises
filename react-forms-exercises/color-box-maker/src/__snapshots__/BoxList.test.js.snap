// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <form>
      <label
        for="width"
      >
        Width (px)
      </label>
      <input
        id="width"
        name="width"
        placeholder="Add Width"
        type="number"
        value=""
      />
      <label
        for="height"
      >
        Height (px)
      </label>
      <input
        id="height"
        name="height"
        placeholder="Add Height"
        type="number"
        value=""
      />
      <label
        for="background"
      >
        Background Color
      </label>
      <input
        id="background"
        name="background"
        placeholder="Add Color"
        type="text"
        value=""
      />
      <button>
        Add Box
      </button>
    </form>
    <div
      class="container"
    >
      <div
        style="width: 200px; height: 100px; background-color: green;"
      />
      <button>
        Remove
      </button>
      <div
        style="width: 100px; height: 200px; background-color: blue;"
      />
      <button>
        Remove
      </button>
    </div>
  </div>
</DocumentFragment>
`;
